DigitalLearningSolutions.Web/Controllers/Support/SupportTicketsController.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace DigitalLearningSolutions.Web.Controllers.Support
+{
+    using System.Threading.Tasks;
+    using DigitalLearningSolutions.Data.Enums;
+    using DigitalLearningSolutions.Data.Helpers;
+    using DigitalLearningSolutions.Web.Attributes;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.Models.Enums;
+    using DigitalLearningSolutions.Web.ViewModels.Support;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.FeatureManagement;
+
+    [SetDlsSubApplication]
+    [SetSelectedTab(nameof(NavMenuTab.Support))]
+    public class SupportTicketsController : Controller
+    {
+        private readonly IConfiguration configuration;
+        private readonly IFeatureManager featureManager;
+
+        public SupportTicketsController(IFeatureManager featureManager, IConfiguration configuration)
+        {
+            this.featureManager = featureManager;
+            this.configuration = configuration;
+        }
+
+        [Route("/{dlsSubApplication}/Support/Tickets")]
+        [Authorize(Policy = CustomPolicies.UserCentreAdminOrFrameworksAdmin)]
+        public async Task<IActionResult> Index(DlsSubApplication dlsSubApplication)
+        {
+            if (!DlsSubApplication.TrackingSystem.Equals(dlsSubApplication) &&
+                !DlsSubApplication.Frameworks.Equals(dlsSubApplication))
+            {
+                return NotFound();
+            }
+
+            var trackingSystemSupportEnabled =
+                DlsSubApplication.TrackingSystem.Equals(dlsSubApplication) &&
+                User.HasCentreAdminPermissions() &&
+                await featureManager.IsEnabledAsync(FeatureFlags.RefactoredTrackingSystem);
+            var frameworksSupportEnabled = DlsSubApplication.Frameworks.Equals(dlsSubApplication) &&
+                                           User.HasFrameworksAdminPermissions();
+
+            if (trackingSystemSupportEnabled || frameworksSupportEnabled)
+            {
+                var model = new SupportTicketsViewModel(
+                    dlsSubApplication,
+                    SupportPage.SupportTickets,
+                    configuration.GetCurrentSystemBaseUrl()
+                );
+                return View("SupportTickets", model);
+            }
+
+            return RedirectToAction("Index", "Home");

```

Comment:
I suspect this will get merged before 608, so can you put a todo here to resolve these by implementing the new attribute style there. I've done the same in ResourcesController for 521 (just merged). We have all the same issues as the other ticket. Incorrect redirect when unauthorized or features aren't enabled, not great bool names (that will be removed) etc.

Reply:
I've added some todos, same as the ones in 521.